frontend/test/parsing/testParseCobegin.cpp

```diff
@@ -106,6 +106,21 @@ static void test1(Parser* parser) {
   assert(cobegin->taskBody(2)->isComment());
 }
 
+static void test2(Parser* parser) {
+  ErrorGuard guard(parser->context());
+  auto parseResult = parseStringAndReportErrors(parser, "test2.chpl",
+      "cobegin with (re A) {;;}\n"
+      "cobegin with () {;;}\n"
+      "cobegin with ref A {;;}\n"
+);
+  auto numErrors = 5;
+  assert(guard.errors().size() == numErrors);
+  assert("invalid intent expression in 'with' clause" == guard.error(1)->message());
+  assert("'with' clause cannot be empty" == guard.error(2)->message());
+  assert("missing parentheses after 'with' clause" == guard.error(4)->message());
+  assert(guard.realizeErrors() == numErrors);
```

Comment:
It would be nice to have a comment describing that the remaining 3 errors are (?)